### PR TITLE
Fix alpha cut-outs and incorrect gamma on web overlays and entities

### PR DESCRIPTION
**Pull request by hyperlogic**
Many websites do not display correctly when used with a Web3DOverlay, or WebEntities.  For example the highfidelity marketplace.  This fixes this by adding a new pixel shader that properly converts from srgb into linear color space and ignores the texel alpha, always writing the overall color alpha value.

### CODE REVIEW

**Comment by hyperlogic** [comment]
Test plan:
   * run this [script](https://gist.githubusercontent.com/hyperlogic/467b213f54bf59a9b933de7047e3d6ee/raw/1528ef343449cebbc7c60e063a343ad1edad341a/web3DOverlayTest.js) you should see the Highfidelity marketplace appear as an overlay.
   * verify that the overlay does not have any transparent holes in it, and has the proper gamma values, i.e. it doesn't look too bright.

File: libraries/render-utils/src/GeometryCache.cpp [hunk 1]
```diff
@@ -1748,6 +1749,46 @@ inline bool operator==(const SimpleProgramKey& a, const SimpleProgramKey& b) {
...
+        slotBindings.insert(gpu::Shader::Binding(std::string("normalFittingMap"), render::ShapePipeline::Slot::MAP::NORMAL_FITTING));
+        gpu::Shader::makeProgram(*_simpleSRGBTexturedUnlitNoDstAlphaShader, slotBindings);
+    });
+
+    if (_simpleSRGBTexturedUnlitNoDstAlphaPipeline) {
+        return _simpleSRGBTexturedUnlitNoDstAlphaPipeline;
+    }
+
+    // If the pipeline did not exist, make it
+    auto state = std::make_shared<gpu::State>();
```
Comment (jherico) [inline libraries/render-utils/src/GeometryCache.cpp]: I don't get why this code (pretty much to the end of the function) isn't inside the `call_once` block

**Comment by hifi-gustavo** [comment]
Build console output for commit: fb84f058d6856784ea5fe8b74b0519c35f23674f available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-fb84f05.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-fb84f05.dmg

File: libraries/render-utils/src/GeometryCache.cpp (hunk 1 above)
Comment (hyperlogic) [inline libraries/render-utils/src/GeometryCache.cpp]: good point, I copied a bad pattern, I moved the pipeline creation into the call_once

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0a35c13a94cd805955b8e24cd4f8dc9f8e0e80dd available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-0a35c13.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-0a35c13.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0a35c13a94cd805955b8e24cd4f8dc9f8e0e80dd available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-0a35c13.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-0a35c13.dmg

**Comment by OmegaHeron** [comment]
Testing Windows 10 64 bit - i5 6500 CPU - GTX970 GPU

Desktop mode looks ok to me - it's solid etc.

hifi-snap-by-omegaheron-on-2016-07-27_13-43-40

DK2 HMD Mode same.

I'm not sure what too bright or too !bright is - it seems ok to my eyes, but, that's purely personal/subjective.

QA Pass

**Comment by ZappoMan** [comment]
This is failing on Mac/Unix...

```
In file included from /data/jenkins/Jenkins/workspace/beta_pr-master/label/ubuntu/libraries/render-utils/src/DebugDeferredBuffer.cpp:21:0:
/data/jenkins/Jenkins/workspace/beta_pr-master/label/ubuntu/libraries/render-utils/src/GeometryCache.h:163:26: error: extra qualification ‘GeometryCache::’ on member ‘getSimpleSRGBTexturedUnlitNoDstAlphaPipeline’ [-fpermissive]
     gpu::PipelinePointer GeometryCache::getSimpleSRGBTexturedUnlitNoDstAlphaPipeline();
                          ^
make[2]: *** [libraries/render-utils/CMakeFiles/render-utils.dir/src/DebugDeferredBuffer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b8e912114deae537794700816f72955b2bf9b8b available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-6b8e912.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-6b8e912.dmg

**Comment by hyperlogic** [comment]
I noticed that web entities are broken as well, so I'm going to fix them as well, then re-mark this as QA Ready

**Comment by hyperlogic** [comment]
Test plan is updated to include a script for testing web entities as well as web overlays.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea310bcc43dc5edce629fd8d5dde629f59e1076a available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-ea310bc.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-ea310bc.dmg

**Comment by OmegaHeron** [comment]
QA Pass

Testing Windows 10 - i5 6500 CPU - GTX970 GPU

Both scripts ran - compared to current RC/DEV entity and overlay now produce correct displays, no transparent areas.  Gamma seems, to my eyes, same as web entity with RC/DEV builds.  Looks correct Desktop and DK2 displays.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e3bc97ef9323a5e9e19d5960fa92897feafa5a3 available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-8e3bc97.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-8e3bc97.dmg

**Comment by hyperlogic** [comment]
some weird jenkins failure

    ERROR: Cannot delete workspace: remote file operation failed: C:\jenkins\workspace\beta_pr-master\label\windows at hudson.remoting.Channel@686c309a:build-10.highfidelity.io: java.nio.file.FileSystemException: C:\jenkins\workspace\beta_pr-master\label\windows\build\ext\vc12\boost\project\src\boost-stamp\boost-download-err.log: The process cannot access the file because it is being used by another process.

test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e3bc97ef9323a5e9e19d5960fa92897feafa5a3 available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-8e3bc97.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-8e3bc97.dmg

**Comment by hyperlogic** [comment]
another jenkins failure

       "C:\jenkins\workspace\beta_pr-master\label\windows\build\PACKAGE.vcxproj" (default target) (1) ->
       "C:\jenkins\workspace\beta_pr-master\label\windows\build\ALL_BUILD.vcxproj" (default target) (2) ->
       "C:\jenkins\workspace\beta_pr-master\label\windows\build\libraries\entities-renderer\entities-renderer.vcxproj" (default target) (29) ->
       "C:\jenkins\workspace\beta_pr-master\label\windows\build\ext\vc12\polyvox\polyvox.vcxproj" (default target) (50) ->
       (CustomBuild target) -> 
         C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.CppCommon.targets(170,5): error MSB6006: "cmd.exe" exited with code 1. [C:\jenkins\workspace\beta_pr-master\label\windows\build\ext\vc12\polyvox\polyvox.vcxproj]

test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e3bc97ef9323a5e9e19d5960fa92897feafa5a3 available at: https://gist.github.com/0b12d2f2cac0c310ea060363838638f2<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-8e3bc97.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8317-8e3bc97.dmg

**Comment by OmegaHeron** [comment]
Just in case it was lost in flurry of Jenkins fails - I did a QA Pass on this above.  Only changes since were not of consequence to that and getting Jenkins to play along.

**Comment by hyperlogic** [comment]
@OmegaHeron Hey, there was an additional change made, which made web entities work instead of just web overlays.  It's worth going thru the test plan again.

**Pull request merged**